x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/profile/SearchProfilesRequest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.security.action.profile;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.Nullable;
+
+import java.io.IOException;
+import java.util.Set;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+public class SearchProfilesRequest extends ActionRequest {
+
+    private final Set<String> dataKeys;
+    @Nullable
+    private final String query;
+    private final int size;
+
+    public SearchProfilesRequest(Set<String> dataKeys, @Nullable String query, @Nullable Integer size) {
```

Comment:
I am actually leaning towards the opposite. That is, let the class's constructor deal with `null` values so that the behaviour is implemented in one place and callers do not have to worry about it. 
That said, this class is really just used by a single caller. So I have no issue following your suggestion.